src/message.coffee

```diff
@@ -33,22 +35,25 @@ class SlackTextMessage extends TextMessage
   # text       - The parsed message text
   # rawText    - The unparsed message text
   # rawMessage - The Slack Message object
-  constructor: (@user, text, rawText, @rawMessage, channel, robot_name) ->
+  constructor: (@user, text, rawText, @rawMessage, channel, robot_name, client) ->
```

Comment:
rather than saving a reference to the client and then deleting it later (good call: don't want reference cycles), how about passing the client into the `buildText()` method? it could be `buildText(client, cb)`.